mutex.go

```
@@ -0,0 +1,108 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package mutex
+
+import (
+	"regexp"
+	"time"
+
+	"github.com/juju/errors"
+)
+
+var (
+	validName = regexp.MustCompile("^[a-zA-Z][a-zA-Z0-9-]*$")
+)
+
+// Releaser defines the Release method that is the only thing that can be done
+// to a acquired mutex.
+type Releaser interface {
+	// Release makes the mutex available for others and invalidates the instance.
+	// Subsequent calls to Release will error.
+	Release() error
```

That makes me pretty uncomfortable. Sure, ignore the error if you want but not giving the caller a choice on what should happen (even if only to log the problem and then exit) seems dicey to me.